### ISSUE TITLE
Add a .keep file in the api_definitions folder when new project is created with init

### ISSUE THREAD

**Issue by fjunior87** [wso2/product-microgateway#1194]
### Describe your problem(s)
When creating a new project using the micro-gw init command it is not adding a .keep file in the api_definitions folder. It seems this folder is required during the build of the micro-gw and when we dont have a file inside a folder, that folder is not sent to git when we check this code into a git repository.
This is a repo tree for a newly created project.
```
abc
├── api_definitions
├── conf
│   └── deployment-config.toml
├── extensions
│   ├── extension_filter.bal
│   ├── startup_extension.bal
│   └── token_revocation_extension.bal
├── grpc_definitions
├── interceptors
│   └── .keep
├── lib
│   └── .keep
└── policies.yaml
```
I used micro gateway toolkit 3.1.0.

### Describe your solution
Update the init command code to add this .keep file inside the api_definitions folder as it is required during the build.

### How will you implement it

---

**Comment by praminda** [comment]
Problem is even if we add a `.keep` file to make the directory identifiable for `git` it will not make any difference to the person whose trying to build the checkedout project. He will get the same error saying api definitions is empty and nothing to build.

Just need to clarify one thing. This means you are pushing and empty (in other terms, half initialized) project into source control. I'm be curious to know the usecase.

**Comment by fjunior87** [comment]
Hi Praminda,

I completely agree, it should have a very clear error message. Right now, it does give much clue on the build errors.
In my scenario, I had a micro-gw project initialized and imported an API from API Manager. So, for that it creates a folder gen/api_definitions and places there the yaml file it uses to build the project.

So, I was checking into git this project, but when building in the CI server it was failing and after some investigation I noticed that the issue was the missing api_definitions folder even we had an api definition inside the gen folder.

**Comment by praminda** [comment]
@fjunior87 Sorry for the late reply. Got your point. We'll look into this. Thanks for bringing this up.

**Comment by Rajith90** [comment]
@fjunior87 , when the api definitions folder and grpc services folder is missing  toolkit give the following error message. Your concern is to have a clear error message than this and also add .keep files to avoid this error

```
[ERROR]
micro-gw: Nothing to build. API definitions/ Grpc Service definitions does not exist.
```

**Comment by fjunior87** [comment]
@Rajith90 in a scenario like I mentioned above:

1. init a micro-gw project;
2. Import an API from API Manager, this will create the gen/api_definitions folder with the yaml;
3. Commit push the project to a git repo;
4. Clone the project in another folder and try to build it;

This will give a generic error with no clue on what is going on. When I did some comparison I noticed that because the api_definitions folder has no file on it, it is not added to the git repository, so its build will fail when it is cloned later. That is the why I am suggesting to add the .keep file under api_definitions.

**Comment by Rajith90** [comment]
@fjunior87 , thanks for the detailed explanation. I got the point now. 
First of all this seems a bug, when gen folder is there the project should be able to build even without api_definitions. We will fix this properly with 3.1.1

**Comment by fjunior87** [comment]
thanks!

**Comment by praminda** [comment]
@fjunior87 We fixed the issue with #1253. We did two things with the fix.
1. Added a `.keep` file to `api_definitions` so that complete standard project structure will always be commited to git, even if the project doesn't include dev first APIs.
1. Removed the mandatory requirement of having `api_definitions` directory in the project structure.
